### DISCUSSION TITLE
Can I reduce fuzziness?

### DISCUSSION THREAD

**Discussion by tmcw** [saghen/blink.cmp#279] [Q&A]
https://github.com/user-attachments/assets/de28f41c-4c71-45e6-b443-276535a043cc

Here's a runthrough of using blink.cmp on a file - these fuzzy suggestions are too fuzzy for me: I want to type `Comment<CR>`, but `blink.nvim` is recommending `CommandMenu` and then the `<CR>` accepts the suggestion. Comment and CommandMenu on a perfix basis are at least 2 character substitutions away, and `Exca` matches `flex-col`, which is also pretty far away from the search string.

Is there a knob that I can turn that will make blink.cmp less permissive in suggesting matches that require fuzziness?

**Comment by saghen** [comment]
Not yet but you could use `fuzzy.use_typo_resistance = false` in the meantime. Just realized that's missing from the README :smile:

**Comment by Parsifa1** [comment]
i'm also a little confused with the sorts：
![telegram-cloud-photo-size-5-6172339188490420030-y](https://github.com/user-attachments/assets/489806df-fb71-4411-a116-e0ba6456c193)
![image](https://github.com/user-attachments/assets/35d0543a-eda7-453b-b984-eb411855db21)

is this was been designed to do so?

**Reply by saghen** [reply]
First screenshot looks like there's a bug with the matched indices but the sorting looks okay, how would you expect it to be sorted? The second screenshot looks bizarre, have you used `iter_swap` in the past? In that case, the frecency would be giving it a bonus. You could test disabling it with `fuzzy.use_frecency = false`

**Reply by Parsifa1** [reply]
as for `sort`, i think the top match will be `std::sort` ...
i never used iter_swap and when `use_frecency` are false it still on the top.
or maybe i have some wrong config?

![image](https://github.com/user-attachments/assets/aaf9ad07-bedf-4daa-a6bb-c501033e8bbc)

**Reply by saghen** [reply]
> as for `sort`, i think the top match will be `std::sort` ...

Ah yeah, that makes sense. I've added `:` as a delimiter to frizbee so this should be fixed in the next release

> i never used iter_swap and when `use_frecency `are false it still on the top.

Could you send one of the `item`s that your LSP returns (i.e. by putting a `vim.print(resolved_item)` in `accept/init.lua`? My best guess is that it's returning a space at the beginning of the item which would break the bonus we give to prefix matches

**Reply by Parsifa1** [reply]
```
06:39:25 PM msg_show {
06:39:25 PM msg_show   client_id = 1,
06:39:25 PM msg_show   cursor_column = 8,
06:39:25 PM msg_show   detail = "void",
06:39:25 PM msg_show   documentation = {
06:39:25 PM msg_show     kind = "markdown",
06:39:25 PM msg_show     value = "From `<algorithm>`"
06:39:25 PM msg_show   },
06:39:25 PM msg_show   filterText = "iter_swap",
06:39:25 PM msg_show   insertText = "std::iter_swap",
06:39:25 PM msg_show   insertTextFormat = 1,
06:39:25 PM msg_show   kind = 3,
06:39:25 PM msg_show   label = " std::iter_swap",
06:39:25 PM msg_show   labelDetails = {
06:39:25 PM msg_show     detail = "(const common_iterator<It, Sent> &x, const common_iterator<It2, Sent2> &y)"
06:39:25 PM msg_show   },
06:39:25 PM msg_show   score = 0.45299568772316,
06:39:25 PM msg_show   score_offset = 0,
06:39:25 PM msg_show   sortText = "41608ba7iter_swap",
06:39:25 PM msg_show   source_id = "lsp",
06:39:25 PM msg_show   source_name = "LSP",
06:39:25 PM msg_show   textEdit = {
06:39:25 PM msg_show     newText = "std::iter_swap",
06:39:25 PM msg_show     range = {
06:39:25 PM msg_show       ["end"] = {
06:39:25 PM msg_show         character = 8,
06:39:25 PM msg_show         line = 45
06:39:25 PM msg_show       },
06:39:25 PM msg_show       start = {
06:39:25 PM msg_show         character = 4,
06:39:25 PM msg_show         line = 45
06:39:25 PM msg_show       }
06:39:25 PM msg_show     }
06:39:25 PM msg_show   }
06:39:25 PM msg_show }
```
this ?

**Reply by Parsifa1** [reply]
and the `std::swap` one:
```
06:45:39 PM msg_show {
06:45:39 PM msg_show   client_id = 1,
06:45:39 PM msg_show   cursor_column = 8,
06:45:39 PM msg_show   detail = "_Require<__not_<__is_tuple_like<Tp>>, is_move_constructible<Tp>, is_move_assignable<Tp>>",
06:45:39 PM msg_show   documentation = {
06:45:39 PM msg_show     kind = "markdown",
06:45:39 PM msg_show     value = "From `<utility>`  \n@endcond"
06:45:39 PM msg_show   },
06:45:39 PM msg_show   filterText = "swap",
06:45:39 PM msg_show   insertText = "std::swap",
06:45:39 PM msg_show   insertTextFormat = 1,
06:45:39 PM msg_show   kind = 3,
06:45:39 PM msg_show   label = " std::swap",
06:45:39 PM msg_show   labelDetails = {
06:45:39 PM msg_show     detail = "(Tp &, Tp &)"
06:45:39 PM msg_show   },
06:45:39 PM msg_show   score = 0.51370817422867,
06:45:39 PM msg_show   score_offset = 0,
06:45:39 PM msg_show   sortText = "407c7d9fswap",
06:45:39 PM msg_show   source_id = "lsp",
06:45:39 PM msg_show   source_name = "LSP",
06:45:39 PM msg_show   textEdit = {
06:45:39 PM msg_show     newText = "std::swap",
06:45:39 PM msg_show     range = {
06:45:39 PM msg_show       ["end"] = {
06:45:39 PM msg_show         character = 8,
06:45:39 PM msg_show         line = 45
06:45:39 PM msg_show       },
06:45:39 PM msg_show       start = {
06:45:39 PM msg_show         character = 4,
06:45:39 PM msg_show         line = 45
06:45:39 PM msg_show       }
06:45:39 PM msg_show     }
06:45:39 PM msg_show   }
06:45:39 PM msg_show }


```

**Reply by Parsifa1** [reply]
> and the `std::swap` one:

updated.

**Reply by saghen** [reply]
Super helpful, thanks! I've just pushed a change on `main` so that it should match on the `filterText` property and it'll treat `:` as a delimiter

**Reply by Parsifa1** [reply]
i tested the main，seems that things are not going well.
anyway，it's a great cmp plugins, i hope it can catch nvim-cmp and then transcend it.

**Reply by saghen** [reply]
Can you elaborate? Did you make sure `cargo build --release` ran?

And I hope so too :)

**Reply by Parsifa1** [reply]
`sort` become better, but i think `std::sort` should higher than `is_sorted`, 'cause `std::sort` are full matched.
![image](https://github.com/user-attachments/assets/9870aae1-6204-4235-9dce-79a479b491d2)

`swap` still have some problem:
![image](https://github.com/user-attachments/assets/bdee0ea4-7e51-4698-9424-e26be9aa345b)

thanks for you effort :)

**Reply by Parsifa1** [reply]
after i actually do some choice to `std::swap`, it seems that usable
![image](https://github.com/user-attachments/assets/f37d58fb-875f-42eb-93ba-c765ab1d8345)
seems that my problem have been solved.
but i think the default order must be adjusted to a more active attitude towards the "full matched"

**Reply by Parsifa1** [reply]
interesting...
![image](https://github.com/user-attachments/assets/7aba4fd2-ab3c-43eb-9258-8e0d909a186f)
![image](https://github.com/user-attachments/assets/82a1d26b-8d26-4853-9c35-0d1a492b38e2)

**Reply by Parsifa1** [reply]
`rangessort` should catch `std::ranges::sort` but not.

**Reply by Parsifa1** [reply]
![image](https://github.com/user-attachments/assets/5274b0de-6dda-4fc5-bf93-c82daff972a8)

![image](https://github.com/user-attachments/assets/534d0d07-e04e-4598-9810-1ef6a6ac84af)

i think the strange sort has been fixed!
many thanks for you effort! ♥️
